cron/loginstats.go

```diff
@@ -0,0 +1,52 @@
+package cron
+
+import (
+	"database/sql"
+)
+
+type LoginCount struct {
+	Count int
+}
+
+// /logins/distinct
+func GetDistinctLoginCount(db *sql.DB, startDate string, endDate string) (LoginCount, error){
+	var count int
+
+	query := `SELECT count(distinct user_id) FROM logins WHERE login_time >= ($1 :: DATE )
+		AND login_time <= ($2 :: DATE) + INTERVAL '1 day';`
+
+	row := db.QueryRow(query, startDate, endDate)
+
+	var logins LoginCount
+	logins = LoginCount{0}
+	err := row.Scan(&count)
+
+	if err != nil {
+		return LoginCount{-1}, err
+	}
+	logins.Count = count
```

Comment:
You could reduce the amount of code here a little bit by scanning directly into the structure. something like this should work:

```
var logins LoginCount
err := row.Scan(&logins.Count)
if err != nil {
    return logins, err
}
```

This change is completely optional, but I thought I'd mention it in case you liked it better. This suggestion applies to `GetLoginCount` as well.

Reply:
making this change actually broke the endpoint. I am not sure if there was another change that had to be made somewhere else, but I am leaving it the way it was for now.

Reply:
I wouldn't have expected another change to be necessary, but it was only a minor suggestion. Leaving the code the way it is seems fine to me.